### PR TITLE
Update the README

### PR DESCRIPTION
**Pull request by 50Wliu**
This can almost certainly be improved but I just wanted to get this out there first in order to receive feedback.

TODO:
* [ ] FAQ:question:
* [x] Improve the Contributing section maybe (and/or improve the guides themselves)
* [x] Update the picture (maybe with a GIF? :movie_camera:)

Part of an ongoing effort to improve Atom's READMEs.

/cc @atom/issue-triage, @atom/feedback

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+* `settings-view:show-keybindings` - Opens up the _Keybindings_ section within the Settings View
+* `settings-view:uninstall-packages` - Opens up the _Packages_ section within the Settings View
+* `settings-view:change-themes` - Opens up the _Themes_ section within the Settings View
+* `settings-view:uninstall-themes` - Opens up the _Themes_ section within the Settings View
+* `settings-view:check-for-updates` - Opens up the _Updates_ section within the Settings View
+* `settings-view:install-packages-and-themes` - Opens up the _Install_ section within the Settings View
+
+### Keybindings
+<kbd>⌘+,</kbd> (Mac) or <kbd>Ctrl+,</kbd> (Windows/Linux) - `settings-view:open`
+Custom keybindings can easily be added by referencing the above commands.
```
Comment (mnquintana) [inline README.md]: Should this be on a separate line? It's currently smashing into the above line.

**Comment by mnquintana** [comment]
Looks good! Mind adding a dependency status badge? (See https://github.com/atom/atom/pull/6457)

File: README.md [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
+### Keybindings
+|Command|Windows|Mac|Linux|
+|-------|-------|---|-----|
+|`settings-view:open`|<kbd>Ctrl+,</kbd>|<kbd>⌘+,</kbd>|<kbd>Ctrl+,</kbd>|
+Custom keybindings can easily be added by referencing the above commands.
+
+### Contributing
+Want to help out?  Here's a helpful [starting point](https://github.com/atom/atom/blob/master/CONTRIBUTING.md) (you might also want to check out the [contributing to packages guide](https://github.com/atom/atom/blob/master/docs/contributing-to-packages.md) for some extra information).
+
+![Settings View](https://cloud.githubusercontent.com/assets/1476/5624241/d29ffbe6-9516-11e4-8bfe-e91623977bc7.png)
```
Comment (mnquintana) [inline README.md]: Why is the image down here now?

Comment (izuzak) [inline README.md]: :+1: I'd love to see it higher up as well, maybe right above the Usage section?

Also, would be great to update the screenshot: use the latest version of Atom, use the default theme, and crop it vertically a bit (reduce the height)? If you feel inspired :performing_arts:, might even make a GIF which shows config settings, installing a package and then configuring the package.

File: README.md [hunk 3]
```diff
@@ -1,7 +1,35 @@
-# Settings View package [![Build Status](https://travis-ci.org/atom/settings-view.svg?branch=master)](https://travis-ci.org/atom/settings-view)
+# Settings View [![Build Status](https://travis-ci.org/atom/settings-view.svg?branch=master)](https://travis-ci.org/atom/settings-view)
 
-Edit config settings, install packages, and change themes in Atom.
+Edit config settings, install packages, and change themes all from within Atom.
```
Comment (izuzak) [inline README.md]: What do you think about adding two more words just to make things a bit clearer?

> Edit core config settings, install and configure packages, and change themes all from within Atom.

File: README.md [hunk 4]
```diff
@@ -1,7 +1,35 @@
...
 
-Open by pressing `cmd-,` or selecting the _Atom > Preferences..._ menu.
+## Usage
+You can open the Settings View by navigating to _Atom > Preferences_ (Mac), _File > Preferences_ (Windows), or _Edit > Preferences_ (Linux).
 
-![screen shot 2015-01-05 at 8 08 18 pm](https://cloud.githubusercontent.com/assets/1476/5624241/d29ffbe6-9516-11e4-8bfe-e91623977bc7.png)
+In order to install new packages and themes, simply click on the _Install_ section located on the left-hand side.
```
Comment (izuzak) [inline README.md]: Maybe :scissors: "simply" and "located"? I think the sentence doesn't lose any meaning without those.

File: README.md [hunk 5]
```diff
@@ -1,7 +1,35 @@
...
+Once installed, community packages/themes (and their settings) are housed within their respective section.
+All packages/themes that have updates will be listed under the _Updates_ section.  Finally, all keybindings (including ones that community packages have added) are available in the _Keybindings_ section.
+
+Want to learn more?  You may want to check the [Getting Started: Atom Basics](https://atom.io/docs/latest/getting-started-atom-basics#settings-and-preferences) and [Using Atom: Atom Packages](https://atom.io/docs/latest/using-atom-atom-packages) sections in the flight manual.
```
Comment (izuzak) [inline README.md]: > You may want to check the ...

Maybe just

> Check out the ...

File: README.md [hunk 6]
```diff
@@ -1,7 +1,35 @@
...
+
+### Commands
+* `settings-view:open` - Opens up the Settings View
+* `settings-view:show-keybindings` - Opens up the _Keybindings_ section within the Settings View
+* `settings-view:uninstall-packages` - Opens up the _Packages_ section within the Settings View
+* `settings-view:change-themes` - Opens up the _Themes_ section within the Settings View
+* `settings-view:uninstall-themes` - Opens up the _Themes_ section within the Settings View
+* `settings-view:check-for-updates` - Opens up the _Updates_ section within the Settings View
+* `settings-view:install-packages-and-themes` - Opens up the _Install_ section within the Settings View
```
Comment (izuzak) [inline README.md]: What do you think about using "Shows" instead of "Opens up", and "of Settings View" instead of "within the Settings View"? :thought_balloon:

Comment (mnquintana) [inline README.md]: What about just "Opens"?

File: README.md [hunk 7]
```diff
@@ -1,7 +1,35 @@
...
+
+### Keybindings
+|Command|Windows|Mac|Linux|
+|-------|-------|---|-----|
+|`settings-view:open`|<kbd>Ctrl+,</kbd>|<kbd>⌘+,</kbd>|<kbd>Ctrl+,</kbd>|
+Custom keybindings can easily be added by referencing the above commands.
+
+### Contributing
+Want to help out?  Here's a helpful [starting point](https://github.com/atom/atom/blob/master/CONTRIBUTING.md) (you might also want to check out the [contributing to packages guide](https://github.com/atom/atom/blob/master/docs/contributing-to-packages.md) for some extra information).
```
Comment (izuzak) [inline README.md]: A bit shorter version:

> Want to help or provide feedback? [Atom's contributing guide](https://github.com/atom/atom/blob/master/CONTRIBUTING.md) will get you started and the [guide for contributing to packages](https://github.com/atom/atom/blob/master/docs/contributing-to-packages.md) has some extra information.

I also wanted to say that feedback is also appreciated -- some users think that the only way to contribute is through code, but that's not true. 

I'd also :heart: to _invite_ users to contribute, instead of _asking_ them if they want to, but can't find the right words right now.

**Comment by izuzak** [comment]
@50Wliu This is a great start! :zap: :zap: 

I'm wondering if @gjtorikian has some :hourglass: to take a look, his wordsmithing services are invaluable. :heart:

**Comment by 50Wliu** [comment]
@mnquintana and @izuzak: Thanks for the feedback :smile:!  I believe I've addressed all of them now, so if any of you could take another look, that would be :sparkles:.

Also @izuzak what do you think of the Contributing section now?  I tried to reword it as best as I could to fit your suggestions.

File: README.md [hunk 8]
```diff
@@ -1,7 +1,35 @@
...
+* `settings-view:uninstall-packages` - Opens the _Packages_ section of the Settings View
+* `settings-view:change-themes` - Opens the _Themes_ section of the Settings View
+* `settings-view:uninstall-themes` - Opens the _Themes_ section of the Settings View
+* `settings-view:check-for-updates` - Opens the _Updates_ section of the Settings View
+* `settings-view:install-packages-and-themes` - Opens the _Install_ section of the Settings View
+
+### Keybindings
+|Command|Windows|Mac|Linux|
+|-------|-------|---|-----|
+|`settings-view:open`|<kbd>Ctrl-,</kbd>|<kbd>⌘-,</kbd>|<kbd>Ctrl-,</kbd>|
```
Comment (mnquintana) [inline README.md]: Even though I usually prefer the `⌘` char for shortcuts, it's basically illegible inside of a `<kbd>` element so I think we should go with `Cmd` here.

Comment (lee-dohm) [inline README.md]: :+1:

File: README.md [hunk 9]
```diff
@@ -1,7 +1,35 @@
...
+Custom keybindings can easily be added by referencing the above commands.
+
+### Contributing
+Always feel free to help out!  Whether it's [filing bugs and feature requests](https://github.com/atom/settings-view/issues/new) or working on some of the [open issues](https://github.com/atom/settings-view/issues), Atom's [contributing guide](https://github.com/atom/atom/blob/master/CONTRIBUTING.md) will help get you started while the [guide for contributing to packages](https://github.com/atom/atom/blob/master/docs/contributing-to-packages.md) has some extra information.
+
+## Copyright
+Copyright &copy; 2014 by [Github Inc](http://www.github.com). See [the license](LICENSE.md) for more details.
```
Comment (izuzak) [inline README.md]: Github -> GitHub :octocat:

File: README.md (hunk 8 above)
Comment (izuzak) [inline README.md]: :+1:

File: README.md [hunk 10]
```diff
@@ -1,7 +1,35 @@
...
+* `settings-view:change-themes` - Opens the _Themes_ section of the Settings View
+* `settings-view:uninstall-themes` - Opens the _Themes_ section of the Settings View
+* `settings-view:check-for-updates` - Opens the _Updates_ section of the Settings View
+* `settings-view:install-packages-and-themes` - Opens the _Install_ section of the Settings View
+
+### Keybindings
+|Command|Windows|Mac|Linux|
+|-------|-------|---|-----|
+|`settings-view:open`|<kbd>Ctrl-,</kbd>|<kbd>⌘-,</kbd>|<kbd>Ctrl-,</kbd>|
+Custom keybindings can easily be added by referencing the above commands.
```
Comment (izuzak) [inline README.md]: :scissors: "easily", and maybe link to docs on customizing keybindings (e.g. [this](https://atom.io/docs/latest/using-atom-basic-customization#customizing-key-bindings) and [this](https://atom.io/docs/latest/behind-atom-keymaps-in-depth))?

File: README.md (hunk 9 above)
Comment (izuzak) [inline README.md]: The license is MIT (I believe the default for all packages). I think we can shorten this to something like:

> MIT License. See [the license](LICENSE.md) for more details.

This makes it more obvious that it's MIT License so folks don't need to open the file just to see that.

File: README.md [hunk 11]
```diff
@@ -1,7 +1,35 @@
...
-Open by pressing `cmd-,` or selecting the _Atom > Preferences..._ menu.
+![Settings View](https://cloud.githubusercontent.com/assets/1476/5624241/d29ffbe6-9516-11e4-8bfe-e91623977bc7.png)
 
-![screen shot 2015-01-05 at 8 08 18 pm](https://cloud.githubusercontent.com/assets/1476/5624241/d29ffbe6-9516-11e4-8bfe-e91623977bc7.png)
+## Usage
+You can open the Settings View by navigating to _Atom > Preferences_ (Mac), _File > Preferences_ (Windows), or _Edit > Preferences_ (Linux).
+
+In order to install new packages and themes, click on the _Install_ section on the left-hand side.
+Once installed, community packages/themes (and their settings) are housed within their respective section.
+All packages/themes that have updates will be listed under the _Updates_ section.  Finally, all keybindings (including ones that community packages have added) are available in the _Keybindings_ section.
```
Comment (izuzak) [inline README.md]: I noticed that there's a double space here (before "Finally"), but then also noticed that you have each sentence in a separate line, so you probably wanted to do that here as well?

File: README.md [hunk 12]
```diff
@@ -1,7 +1,35 @@
...
 
-Open by pressing `cmd-,` or selecting the _Atom > Preferences..._ menu.
+![Settings View](https://cloud.githubusercontent.com/assets/1476/5624241/d29ffbe6-9516-11e4-8bfe-e91623977bc7.png)
 
-![screen shot 2015-01-05 at 8 08 18 pm](https://cloud.githubusercontent.com/assets/1476/5624241/d29ffbe6-9516-11e4-8bfe-e91623977bc7.png)
+## Usage
+You can open the Settings View by navigating to _Atom > Preferences_ (Mac), _File > Preferences_ (Windows), or _Edit > Preferences_ (Linux).
+
+In order to install new packages and themes, click on the _Install_ section on the left-hand side.
+Once installed, community packages/themes (and their settings) are housed within their respective section.
```
Comment (izuzak) [inline README.md]: I think we can remove the parentheses here and just say "... community packages/themes and their settings are housed...".

File: README.md [hunk 13]
```diff
@@ -1,7 +1,35 @@
...
+* `settings-view:check-for-updates` - Opens the _Updates_ section of the Settings View
+* `settings-view:install-packages-and-themes` - Opens the _Install_ section of the Settings View
+
+### Keybindings
+|Command|Windows|Mac|Linux|
+|-------|-------|---|-----|
+|`settings-view:open`|<kbd>Ctrl-,</kbd>|<kbd>⌘-,</kbd>|<kbd>Ctrl-,</kbd>|
+Custom keybindings can easily be added by referencing the above commands.
+
+### Contributing
```
Comment (mnquintana) [inline README.md]: I think this should be an h2 `##` - it's not really a subsection of `Usage`.

File: README.md [hunk 14]
```diff
@@ -1,7 +1,35 @@
...
+Always feel free to help out!  Whether it's [filing bugs and feature requests](https://github.com/atom/settings-view/issues/new) or working on some of the [open issues](https://github.com/atom/settings-view/issues), Atom's [contributing guide](https://github.com/atom/atom/blob/master/CONTRIBUTING.md) will help get you started while the [guide for contributing to packages](https://github.com/atom/atom/blob/master/docs/contributing-to-packages.md) has some extra information.
```
Comment (mnquintana) [inline README.md]: :fire: Extra space after `help out!`

Comment (mnquintana) [inline README.md]: If we're adding this, should we :fire: CONTRIBUTING.md?

Comment (izuzak) [inline README.md]: > If we're adding this, should we :fire: CONTRIBUTING.md?

The benefit of having CONTRIBUTING.md is that it shows up as a [banner at the top of the page](https://help.github.com/articles/setting-guidelines-for-repository-contributors/) when you create issues and pull requests. I'm pretty sure most folks won't click that link, but for those 1% that will -- it still might be helpful. :smile: I think we should keep it here.

File: README.md (hunk 2 above)
Comment (mnquintana) [inline README.md]: Oh yeah! Forgot about that. We should probably still link to it in this description though?

**Comment by izuzak** [comment]
> Also @izuzak what do you think of the Contributing section now? I tried to reword it as best as I could to fit your suggestions.

@50Wliu Ah, forgot to answer this -- I like the inviting tone of it :+1:, and I don't mind the added length there, to be honest. Personally, I'd be okay with shipping that, but if anyone else has any other wordsmithing advice -- that'd be great. (I'm not a native speaker, and sometimes have trouble finding the right words for these things. :book:)

File: README.md [hunk 15]
```diff
@@ -1,7 +1,35 @@
...
 
-Edit config settings, install packages, and change themes in Atom.
+Edit core config settings, install and configure packages, and change themes all from within Atom.
 
-Open by pressing `cmd-,` or selecting the _Atom > Preferences..._ menu.
+![Settings View](https://cloud.githubusercontent.com/assets/1476/5624241/d29ffbe6-9516-11e4-8bfe-e91623977bc7.png)
 
-![screen shot 2015-01-05 at 8 08 18 pm](https://cloud.githubusercontent.com/assets/1476/5624241/d29ffbe6-9516-11e4-8bfe-e91623977bc7.png)
+## Usage
+You can open the Settings View by navigating to _Atom > Preferences_ (Mac), _File > Preferences_ (Windows), or _Edit > Preferences_ (Linux).
```
Comment (lee-dohm) [inline README.md]: Alphabetic ordering on the platforms?

File: README.md [hunk 16]
```diff
@@ -1,7 +1,35 @@
-# Settings View package [![Build Status](https://travis-ci.org/atom/settings-view.svg?branch=master)](https://travis-ci.org/atom/settings-view)
+# Settings View [![Build Status](https://travis-ci.org/atom/settings-view.svg?branch=master)](https://travis-ci.org/atom/settings-view) [![Dependency Status](https://david-dm.org/atom/settings-view.svg)](https://david-dm.org/atom/atom)
 
-Edit config settings, install packages, and change themes in Atom.
+Edit core config settings, install and configure packages, and change themes all from within Atom.
 
-Open by pressing `cmd-,` or selecting the _Atom > Preferences..._ menu.
+![Settings View](https://cloud.githubusercontent.com/assets/378023/7411007/ecc36a1e-ef6e-11e4-9514-9564801e6cd8.png)
```
Comment (mnquintana) [inline README.md]: Can you use One Dark for the screenshot?

File: README.md (hunk 8 above)
Comment (philipgiuliani) [inline README.md]: :+1:

**Comment by mnquintana** [comment]
@simurai Are there any classes / styling stuff we should document here? (ie. for theme authors)

**Comment by simurai** [comment]
@mnquintana Themes can override stuff if they want, but there shouldn't be anything that is required.

**Comment by lee-dohm** [comment]
@simurai, what I believe @mnquintana is referring to is that one of the things we're trying to do to improve the READMEs is to document high-level classes to give more than just theme authors the opportunity to easily style their environment (such as the person that just wants to increase the font size or change the background color, etc). An example is what I did with my [soft-wrap-indicator package](https://atom.io/packages/soft-wrap-indicator) here:

screen shot 2015-04-30 at 7 28 56 am

With how complex the Settings View is, I think just documenting the topmost couple levels of selectors is good enough to get people started for simple things. Something like:

* `.settings-view` - For styling the Settings View
    * `.settings-view .config-menu` - For styling the left-hand selection bar
    * `.settings-view .panels` - For styling the right-hand viewer section

Maybe you have some better ideas what to name the pieces @simurai?

**Comment by simurai** [comment]
@lee-dohm Oh.. I see. It would also be for customizing in `styles.less`. Ok, yeah, that might be helpful. I'll add something.

**Comment by simurai** [comment]
@lee-dohm Hmm.. it's kinda tricky.. because depending on which properties you wanna style, the selector needs to be different. For example you can't just do `.settings-view { background-color: pink; }` because it gets styled from core with a huge chained selector: `atom-pane-container atom-pane .item-views .pane-item { background-color: #272b33; }`. So to win the specificity fight, we would have to use `atom-pane-container atom-pane .item-views .pane-item.settings-view` which looks kinda ugly in the README. And that selector could change in core without getting updated here.

So instead, I was thinking of showing the DOM tree like:

```
└─.settings-view
  ├─ .config-menu
    ├─ .panels-menu
    └─ .button-area
  └─ .panels
    ├─ .section.general-panel
    ├─ .section.settings-panel
    └─ .section.settings-panel
```

You would get an overview how the markup is structured. But that doesn't solve the above problem and there are actually some wrapper `div`s that are not included, so that might be misleading.

So... this leads to this: How about we just add 1-2 real world examples how to override some more common stuff like:

```css
/* Change the color of the setting titles */
.settings-view .section .section-heading {
  color: pink;
}

/* Change the font size of the setting descriptions */
.settings-view .setting-description {
  font-size: 13px;
}
```

Or things that people keep asking in issue/discuss. And a note that they can use the DevTools to look up other selectors.

**Comment by lee-dohm** [comment]
@simurai Sounds good! I'm fine with just mentioning the common requests. That's all that I've ever intended. This is a README, not a tutorial on CSS :grinning: And I'm still fine with just mentioning the selectors and then perhaps linking to the place in https://atom.io/docs that talks about styling Atom.

**Comment by 50Wliu** [comment]
Just changed the commands list to a table-any feedback?
For this particular README I've included the `Selector` column since all of the commands were under the same selector (and due to the not-so-pretty bug where `atom-workspace` wrapped before the description).

**Comment by 50Wliu** [comment]
Also, it looks like we could manage to combine the Commands and Keybindings sections.

**Comment by 50Wliu** [comment]
Just combined commands and keybindings, but I'm not too sure I like the final result.

Other than that, pretty sure this is good to go if no one has any other feedback.

**Comment by izuzak** [comment]
@50Wliu Sorry for not looking at this sooner -- I'll take a look over the weekend.

File: README.md [hunk 17]
```diff
@@ -1,7 +1,51 @@
...
+  color: white;
+}
+
+// Change the font size of the setting descriptions
+.settings-view .setting-description {
+  font-size: 13px;
+}
+```
+
+Use the [developer tools](https://atom.io/docs/v0.196.0/hacking-atom-creating-a-theme#developer-tools) to find more selectors.
```
Comment (izuzak) [inline README.md]: Use `latest` instead of `0.196.0` here?

**Comment by izuzak** [comment]
> Just combined commands and keybindings, but I'm not too sure I like the final result.

Yeah, I agree -- I liked it more when those were separated. 

Other than that -- I think this looks good to merge. Thanks again, @50Wliu :+1:

**Comment by jlord** [comment]
@izuzak I just took care of that last comment of yours re: the latest docs link :+1: 

So excited about having a better readme! Thanks everyone! :clap:

**Comment by 50Wliu** [comment]
Whoops...totally forgot about this :frowning:.
I'll open up another PR to separate the commands and keybindings again.
